library/alloc/src/collections/btree/borrow.rs

```diff
@@ -0,0 +1,43 @@
+use core::marker::PhantomData;
+use core::ptr::NonNull;
+
+/// Models a reborrow of some unique reference, when you know that the reborrow
+/// and all its descendants (i.e., all pointers and references derived from it)
+/// will not be used any more at some point, after which you want to use the
+/// original unique reference again.
+///
+/// The borrow checker usually handles this stacking of borrows for you, but
+/// some control flows that exhibit this stacking are still too complicated
+/// for the compiler to follow. A `DormantMutRef` allows more than what the
+/// compiler understands, but less than it could, in that you only get back
+/// the original reference twice, once at the start and once at the end.
```

Comment:
The "just twice" comment is confusing to me, what do you mean by that?

Reply:
Not more than twice. Once would mean you might as well use the original reference, twice is the minimum.

Reply:
But that would it even mean to get back the original reference thrice?

Reply:
You can just use the return value from `awaken` and call `DormanRefMut::new` again on it, if you want "more than twice".

Reply:
That's true, I didn't realize you can already do more. In other words, the class is perfect!

Reply:
Just the documentation wasn't...